src/core/mac/mac.cpp

```diff
@@ -1785,6 +1913,9 @@ void Mac::HandleReceivedFrame(RxFrame *aFrame, otError aError)
 
     otDumpDebgMac("RX", aFrame->GetHeader(), aFrame->GetLength());
     Get<MeshForwarder>().HandleReceivedFrame(*aFrame);
+#if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
```

Comment:
I think we can consider removing `#if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE` and do `UpdateIdleMode()` in all situations.